src/main/java/com/mraof/minestuck/item/weapon/PropelEffect.java

```diff
@@ -0,0 +1,75 @@
+package com.mraof.minestuck.item.weapon;
+
+import com.mraof.minestuck.player.EnumAspect;
+import com.mraof.minestuck.player.Title;
+import com.mraof.minestuck.world.storage.ClientPlayerData;
+import com.mraof.minestuck.world.storage.PlayerSavedData;
+import net.minecraft.client.entity.player.ClientPlayerEntity;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.entity.player.ServerPlayerEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.*;
+import net.minecraft.util.math.Vec3d;
+import net.minecraft.world.World;
+
+public class PropelEffect implements ItemRightClickEffect
+{
+	public static final PropelEffect BREATH_PROPEL = new PropelEffect(3, EnumAspect.BREATH);
+	
+	private final double velocity;
+	private final EnumAspect aspect;
+	
+	public PropelEffect(double velocity, EnumAspect aspect)
+	{
+		this.velocity = velocity;
+		this.aspect = aspect;
+	}
+	
+	@Override
+	public ActionResult<ItemStack> onRightClick(World world, PlayerEntity player, Hand hand)
+	{
+		ItemStack itemStack = player.getHeldItem(hand);
+		propelAction(player, itemStack, getVelocityMod(), hand);
+		return ActionResult.resultPass(itemStack);
+	}
+	
+	private double getVelocityMod()
+	{
+		return velocity;
+	}
+	
+	void propelAction(PlayerEntity player, ItemStack stack, double velocity, Hand hand)
+	{
+		Title title = null;
+		if(player instanceof ClientPlayerEntity)
+		{
+			title = ClientPlayerData.getTitle();
+		}
+		
+		if(player instanceof ServerPlayerEntity)
+		{
+			title = PlayerSavedData.getData((ServerPlayerEntity) player).getTitle();
+			if(player.getCooldownTracker().getCooldown(stack.getItem(), 1F) <= 0 && title != null && title.getHeroAspect() == aspect)
+				propelActionSound(player.world, player);
```

Comment:
Might I ask why this is here, and not in the code block below?

Reply:
It is not important, I chose to put it here because I thought the sound only plays on server side

Reply:
I think it might be better code form to have it in the code block below together with the other effects, separated from the code to obtain the title. Though I suppose it's not a big deal, and I'm fine merging it as it is if you don't want to bother with it.

Reply:
I think it helps mildly, in that its obvious in which conditions it would be activating. So I am fine with merging